python/lsst/ap/association/mpSkyEphemerisQuery.py

```diff
@@ -0,0 +1,218 @@
+    )
+
+    ssObjects = connTypes.Output(
+        doc="MPSky-provided Solar System objects observable in this detector-visit",
+        name="preloaded_ssObjects",
+        storageClass="DataFrame",
+        dimensions=("instrument", "group", "detector"),
+    )
+
+
+class MPSkyEphemerisQueryConfig(
+        PipelineTaskConfig,
+        pipelineConnections=MPSkyEphemerisQueryConnections):
+    observerCode = pexConfig.Field(
+        dtype=str,
+        doc="IAU Minor Planet Center observer code for queries "
+            "(Rubin Obs./LSST default is X05)",
+        default='X05'
+    )
+    queryRadiusDegrees = pexConfig.Field(
+        dtype=float,
+        doc="On sky radius for Ephemeris cone search. Defaults "
+            "to the radius of Rubin Obs FoV in degrees",
+        default=1.75
+    )
+    MPSkyURL = pexConfig.Field(
+        dtype=str,
+        doc="URL to query mpsky service",
+        default="http://sdfrome001.sdf.slac.stanford.edu:3666"
+    )
+
+
+class MPSkyEphemerisQueryTask(PipelineTask):
+    """Tasks to query the MPSky service and retrieve the solar system objects
+    that are observable within the input visit.
+    """
+    ConfigClass = MPSkyEphemerisQueryConfig
+    _DefaultName = "MPSkyEphemerisQuery"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        outputs = self.run(**inputs)
+
+        butlerQC.put(outputs, outputRefs)
+
+    @timeMethod
+    def run(self, predictedRegionTime):
+        """Parse the information on the current visit and retrieve the
+        observable solar system objects from MPSky.
+
+        Parameters
+        ----------
+        predictedRegionTime : `pipe.base.utils.RegionTimeInfo`
+        RegionTime of the predicted exposure
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components:
+
+            - ``ssObjects``: `pandas.DataFrame`
+                DataFrame containing Solar System Objects in field of view as
+                retrieved by MPSky. The columns are as follows:
+                ``Name``
+                    object name (`str`)
+                ``ra``
+                    RA in decimal degrees (`float`)
+                ``dec``
+                    DEC in decimal degrees (`float`)
+                ``obj_poly``
+                    DO NOT USE until t_min issue is resolved
+                ``obs_poly``
+                    DO NOT USE until t_min issue is resolved
+        """
+        # Grab the visitInfo from the raw to get the information needed on the
+        # full visit.
+
+        # Midpoint time of the exposure in JD
+        region = predictedRegionTime.region
+        timespan = predictedRegionTime.timespan
+        expCenter = SpherePoint(region.getBoundingCircle().getCenter())
+
+        # Make sure date is non-NaN.
+        expMidPointEPOCH = (timespan.begin.mjd + timespan.end.mjd)/2
```

Comment:
After reviewing #230, I don't think this will work -- `astropy.time.Time` doesn't support the `+` operator. Fortunately, once that PR merges, you should be able to just do:
```suggestion
        expMidPointEPOCH = utils.getMidpointFromTimespan(timespan).mjd
```
As a bonus, `getMidpointFromTimespan` would do the input validation for you.